src/rust/catmem/mod.rs

```diff
@@ -188,14 +188,11 @@ impl CatmemLibOS {
                 match self.qtable.borrow().get(&qd) {
                     Some(queue) => {
                         let pipe: &Pipe = queue.get_pipe();
-                        let coroutine: Pin<Box<Operation>> = if pipe.eof() {
-                            // Handle end of file.
-                            Box::pin(async move {
-                                let cause: String = format!("connection reset (qd={:?})", qd);
-                                error!("pop(): {:?}", &cause);
-                                (qd, OperationResult::Failed(Fail::new(libc::ECONNRESET, &cause)))
-                            })
-                        } else {
+
```

Comment:
Should we return an error here instead of a qt? We know that the coroutine won't complete without an error. Also, it looks like the push coroutine doesn't check for eof, so we might want to avoid scheduling it.

Reply:
I have not followed your suggestion. Could you clarify? Note that

```rust
// We do not check for EoF because pipes are unidirectional,
// and if EoF is set pipe for a push-only pipe, that pipe is closed.
```

Reply:
Should we still be pushing to a closed pipe? Before this change, we scheduled a different coroutine but I'm suggesting that we should skip line 195 and just return an error because the push to a closed pipe won't succeed.

Reply:
1) We should not be pushing to a closed pipe, that is the whole point of the PR.  Also, may I remind you that we have not addressed bugs for Catmem LibOS (like we did in Catnap, when we introduced the Socket abstraction) and this behavior may not be fully functinoal.
2) Line 195 is spawning the usual co-routine, we cannot skip it.
3) I have removed only the code that spawns the co-routine for `ECONNRESET` in `push()`, we still do it on `pop()`

Reply:
yes that was my point. We still create a coroutine here, which we should skip on eof.

Reply:
Basically I’m asking why we don’t return on line 194

Reply:
I have added the unreachable statement, as we agreed offline.